src/components/InternationalizedArray.tsx

```diff
@@ -28,18 +31,24 @@ export default function InternationalizedArray(props: InternationalizedArrayProp
   const readOnly = typeof schemaType.readOnly === 'boolean' ? schemaType.readOnly : false
   const {options} = schemaType
   const toast = useToast()
+  const {value: document} = useFormBuilder()
+  const selectedValue = getSelectedValue(options.select, document)
```

Comment:
Since V3 haven't been optimised its context providers and `useContext` hooks to only rerender when changed (we had to chose between spending time on moving to a React 18 baseline or fix context state, so we went with 18 as it gives us more opportunities to fix context state long term) it's a good idea to add `useDeferredValue` and `useMemo` here to avoid blocking the UI if there's a lot of multiplayer activity happening in a document:
```suggestion
  const {value: document} = useFormBuilder()
  const deferredDocument = useDeferredValue(document)
  const selectedValue = useMemo(() =>
    getSelectedValue(options.select, deferredDocument)
  , [getSelectedValue, options.select, deferredDocument])
```

Reply:
Great suggestion, by doing this we don't even get the loading state anymore after changing languages. It now seems to fetch in the background, which in my opinion makes this change quite useful.

Reply:
Nice, I thought only the `useTransition` and `startTransition` APIs could have that effect 🤯 At the time it felt a bit risky to make React 18 the baseline, but it does start to look like it was the right decision for our ecosystem 😮‍💨 It's just so nice to have more strategies available when dealing with async stuff and adapting to IO constraints 😌